swapi/components/owned-entities-display.js

```diff
@@ -0,0 +1,110 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import ChooseSpaceshipModal from './choose-spaceship';
+
+class OwnedEntitiesDisplay extends Component {
+	static propTypes = {
+		ownedEntities: PropTypes.shape({
+			people: PropTypes.arrayOf(
+				PropTypes.shape({
+					name: PropTypes.string.isRequired
+				})
+			),
+			starships: PropTypes.arrayOf(
+				PropTypes.shape({
+					crew_people: PropTypes.any.isRequired
+				})
+			)
+		}).isRequired
+	};
+
+	state = {
+		showChooser: false,
+		activeShip: {}
+	};
+	handleCloseChooser = () => {
+		this.setState({ showChooser: false, activeShip: {} });
+	};
+	handleOpenChooser = ship => {
+		this.setState({ showChooser: true, activeShip: ship });
+	};
+
+	render() {
+		return (
+			<div>
+				{this.state.showChooser ? (
+					<ChooseSpaceshipModal
+						onCloseClick={this.handleCloseChooser}
+						activeShip={this.state.activeShip}
+					/>
+				) : (
+					this.renderOwnedEntities()
+				)}
+			</div>
+		);
+	}
+
+	renderOwnedEntities() {
+		const { ownedEntities } = this.props;
+		let uniquePeople = countUniques(ownedEntities.people); // We are all unique :)
```

Comment:
`countUniques` implicerer lidt at den returnerer en number, men jeg kan se at `uniquePeople` bliver brugt som object. 

Overvej i øvrigt også, om det ikke burde være in en `mapStateToProps` function denne beregning burde ligge.